src/App.tsx

```diff
@@ -1,26 +1,343 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import { useEffect, FC, useRef, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
+import { USER_ID, addTodo, editTodo, getTodos, removeTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { Status } from './types/Status';
 
-const USER_ID = 0;
+const filterTodos = (todos: Todo[], filter: string) => {
+  switch (filter) {
+    case Status.Active:
+      return todos.filter(todo => !todo.completed);
 
-export const App: React.FC = () => {
+    case Status.Completed:
+      return todos.filter(todo => todo.completed);
+
+    default:
+      return todos;
+  }
+};
+
+export const App: FC = () => {
+  const titleField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>();
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [filter, setFilter] = useState(Status.All);
+  const [unCompletedTodos, setUncompletedTodos] = useState(0);
+  const [allIsCompleted, setAllIsCompleted] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [allCompletedInProcess, setAllCompletedInProcess] = useState(false);
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  useEffect(() => {
+    setErrorMessage('');
+    getTodos()
+      .then(data => setTodos(data))
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+      });
+  }, []);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  }, [errorMessage]);
+
+  useEffect(() => {
+    let allUnCompletedTodos = 0;
+
+    todos.forEach(todo => {
+      allUnCompletedTodos += todo.completed ? 0 : 1;
+    });
+    setUncompletedTodos(allUnCompletedTodos);
+
+    setAllIsCompleted(allUnCompletedTodos === 0);
+  }, [todos]);
```

Comment:
do you really need a `useEffect` for it? think about to use useMemo:
```ts
const [uncompletedTodos, allIsCompleted] = useMemo(() => {
  const allUnCompletedTodos = todos.reduce((count, todo) => todo.completed ? count : count + 1, 0);

  return [allUnCompletedTodos, allUnCompletedTodos === 0];
}, [todos]);
```

after it you can remove the states